Operator installation guide (#11309)
keycloak/keycloak: docs/guides/src/main/operator/installation.adoc

<@tmpl.guide
title="Keycloak Operator Installation"
summary="How to install the Keycloak Operator on Kubernetes and OpenShift">

== Keycloak Operator Installation
In this guide we will show how to install the Keycloak Operator in your Kubernetes or OpenShift cluster.

=== OLM Installation

> You can install the Keycloak Operator in Kubernetes environment is to use Operator Lifecycle Manager (OLM).

Reviewer: ```suggestion
The recommended way to install the Keycloak Operator in Kubernetes environments is to use the Operator Lifecycle Manager (OLM).
```

Reviewer: plus perhaps a Link to OLM page.

Author: Fixed in a different way and added the link

> Before moving on, make sure you have OLM installed, if not follow this https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/install/install.md#install-a-release[guide].

Reviewer: ```suggestion

==== Prerequisites
Make sure OLM is installed in your environment. For Guidance on how to install OLM, follow this https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/install/install.md#install-a-release[guide].
```

Reviewer: perhaps also add more prerequisites, like kubectl etc?

> To install the Keycloak Operator follow the https://olm.operatorframework.io/docs/tasks/install-operator-with-olm/[instructions].

Reviewer: ```suggestion
The Keycloak Operator OLM package can be installed from the OLM catalog. For general instructions on how to install operators using OLM, follow the https://olm.operatorframework.io/docs/tasks/install-operator-with-olm/[instructions] on the OLM page.
```

Reviewer: Not sure if we should not add a rough doc on our side though, as we'Re pointing to OLM often. Then again we would have to maintain changes on their side, so works for me.

> In the default Catalogs the Keycloak Operator is named `keycloak-operator` and make sure to use the `candidate` channel.

Reviewer: ```suggestion
In the default Catalog, the Keycloak Operator is named `keycloak-operator`. Make sure to use the `candidate` channel to find the operator.
```

Reviewer: Why do we need to use the "candidate" channel? If this is a temporary thing, perhaps "to find the keycloak operator candidate release."?

Reviewer 2: The `candidate` channel is currently required to comply with the [recommended naming](https://olm.operatorframework.io/docs/best-practices/channel-naming/#recommended-channel-naming). We will probably move to `stable` in KC 20.

==== OpenShift UI

> On OpenShift you can use a conveninet UI on top of OLM.

Reviewer: ```suggestion
On OpenShift, use the built-in OLM UI to install the Keycloak Operator.
```

> Navigate `Home` → `Operators` → `OperatorHub` in the menu on the left side of the OpenShift Console.

Reviewer: ```suggestion
Navigate to `Home` → `Operators` → `OperatorHub` using the menu on the left side of the OpenShift Console.
```

> Then, focus on the search input box and type "keycloak":

Reviewer: ```suggestion
Search for "keycloak" on the search input box:
```

image::{generatedGuideImages}/select-operator.jpeg["Select the Keycloak Operator in the UI"]

> Next, navigate to Keycloak Operator and click on it. Next, follow the instructions on the screen, make sure you are installing from the `candidate` channel:

Reviewer: ```suggestion
Select the Keycloak Operator from the list of results. After that, follow the instructions on the screen. Make sure you are installing from the `candidate` channel:
```

Reviewer: not sure if the "candidate" sentence and pic should come before this, because when reading it it may be that i would have followed instructions on the screen to come back to this guide and only later on see that i installed from wrong channel. makes sense for you?

image::{generatedGuideImages}/configure-operator.jpeg["Configure Keycloak Operator"]

=== Vanilla Kubernetes Installation

> To install the operator on a vanilla Kubernetes cluster, you need first to install the CRDs:

Reviewer: ```suggestion
To install the operator on a vanilla Kubernetes cluster, you first need to install its CRDs by running the following commands:
```

[source,bash,subs="attributes+"]
----
kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/{version}/kubernetes/keycloaks.k8s.keycloak.org-v1.yml
kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/{version}/kubernetes/keycloakrealmimports.k8s.keycloak.org-v1.yml
----

> And then install the Keycloak Operator deployment:

Reviewer: ```suggestion
After successfull CRD installation, install the Keycloak Operator deployment by running the following command:
```
